pkg/credentialwatcher/secretwatcher.go

```diff
@@ -123,6 +123,12 @@ func (s *secretWatcher) updateAccountRotateCredentialsStatus(log logr.Logger, ac
 		return
 	}
 
+	// Only rotate STS credentials if the account CR is in a Ready state
+	if accountInstance.Status.State != string(awsv1alpha1.AccountReady) {
```

Comment:
Does `string(awsv1alpha1.AccountReady)` have to be wrapped in `string()`?  It's already a string, isn't it?

Reply:
It should be fixed as its actually of type `AccountConditionType`